<?xml version="1.0"?>
<launch>
    <!-- declare arg to be passed in -->
    <arg name="me" default="ally1" />
    <arg name="ally" default="ally2" />
    <arg name="robot_number" default="1" />
    <!-- <arg name="sim_mode" default="false" /> -->
    <arg name="team_side" default="home" />
    <arg name="vel_cmds_remap" default="vel_cmds" />

    <!-- using which side of the field we are on, determine the appropriate -->
    <!-- connections to the output of the vision system -->
    <arg name="other_side" value="away" if="$(eval team_side == 'home')" />
    <arg name="other_side" value="home" if="$(eval team_side == 'away')" />

    <!-- Hardware agnostic code -->
    <node name="ally1_estimator" pkg="slash_dash_bang_hash" type="Estimator">
      <remap from="ally1_vision" to="/vision/$(arg team_side)1" />
      <remap from="ally2_vision" to="/vision/$(arg team_side)2" />
      <remap from="opponent1_vision" to="/vision/$(arg other_side)1" />
      <remap from="opponent2_vision" to="/vision/$(arg other_side)2" />
      <remap from="ball_vision" to="/vision/ball" />
    </node>
    <node name="AI" pkg="slash_dash_bang_hash" type="AI" output="screen">
    </node>
    <node name="path_planner" pkg="slash_dash_bang_hash" type="PathPlanner" output="screen">
      <remap from="destination" to="/slash_dash_bang_hash_$(arg team_side)/$(arg me)/$(arg me)_destination" />
      <remap from="desired_pose" to="/slash_dash_bang_hash_$(arg team_side)/$(arg me)/$(arg me)_desired_pose" />
    </node>
    <node name="controller" pkg="slash_dash_bang_hash" type="Controller" output="screen">
        <!-- The controller needs to know which robot state to listen to -->
        <remap from="robot_state" to="/slash_dash_bang_hash_$(arg team_side)/$(arg me)/$(arg me)_state" />
        <remap from="desired_pose" to="/slash_dash_bang_hash_$(arg team_side)/$(arg me)/$(arg me)_desired_pose" />
        <remap from="vel_command" to="/$(arg team_side)$(arg robot_number)/command" />
    </node>

</launch>
